sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/_base.py

```diff
@@ -199,6 +199,20 @@ def _handle_transmit_from_storage(self, envelopes: List[TelemetryItem], result:
             elif result == ExportResult.SUCCESS:
                 # Try to send any cached events
                 self._transmit_from_storage()
+
+            # If filesystem is readonly, track dropped items in customer statsbeat
+            if self.storage.filesystem_is_readonly:
```

Comment:
I think it is a bit risky to depend on the state of the `self.storage` with a stored static value within `self` since attributes like `persistent_storage_full` can be dynamic and change over time. We should:

1. Keep `self.filesystem_is_readonly` since that will never change.
2. Move the sdkstats `track` logic to `_transmit_from_storage` from when we actually attempt to write to storage.
3. Modify `LocalFileStorage.put` to return different statuses depending on the outcome. We will have to modify some of the try: excepts like in `LocalFileBlob.put` to not simply just be silent to be able to capture exceptions being thrown.
4. Depending on the status of `LocalFileStorage.put` returned, call the corresponding `track`.